src/azure-cli/azure/cli/command_modules/network/_validators.py

```diff
@@ -480,12 +480,12 @@ def validate_servers(namespace):
 
 
 def validate_subresource_list(cmd, namespace):
-    if namespace.target_resources:
+    if namespace.parameters:
         SubResource = cmd.get_models('SubResource', resource_type=ResourceType.MGMT_NETWORK_DNS)
         subresources = []
-        for item in namespace.target_resources:
+        for item in namespace.parameters:
             subresources.append(SubResource(id=item))
-        namespace.target_resources = subresources
+        namespace.parameters = subresources
```

Comment:
When using inherent way, such validator is not valid anymore also related codes in _params.py should be removed.

Reply:
removed validator and params for list-references